network/wsPeer.go

```diff
@@ -384,42 +442,51 @@ func (wp *wsPeer) readLoop() {
 			return
 		}
 
-		// Handle Topic message
-		if msg.Tag == protocol.TopicMsgRespTag {
-			topics, err := UnmarshallTopics(msg.Data)
-			if err != nil {
-				wp.net.log.Warnf("wsPeer readLoop: could not read the message from: %s %s", wp.conn.RemoteAddr().String(), err)
-				continue
-			}
-			requestHash, found := topics.GetValue("RequestHash")
-			if !found {
-				wp.net.log.Warnf("wsPeer readLoop: message from %s is missing the RequestHash", wp.conn.RemoteAddr().String())
-				continue
-			}
-			hashKey, _ := binary.Uvarint(requestHash)
-			channel, found := wp.getAndRemoveResponseChannel(hashKey)
-			if !found {
-				wp.net.log.Warnf("wsPeer readLoop: received a message response from %s for a stale request", wp.conn.RemoteAddr().String())
-				continue
-			}
-			resp := Response{Topics: topics}
-
-			select {
-			case channel <- &resp:
-			default:
-				wp.net.log.Warnf("wsPeer readLoop: channel blocked. Could not pass the response to the requester", wp.conn.RemoteAddr().String())
-			}
-		} else {
-			select {
-			case wp.net.readBuffer <- msg:
-			case <-wp.closing:
-				wp.net.log.Debugf("peer closing %s", wp.conn.RemoteAddr().String())
-				return
-			}
+		select {
+		case wp.net.readBuffer <- msg:
+		case <-wp.closing:
+			wp.net.log.Debugf("peer closing %s", wp.conn.RemoteAddr().String())
+			return
 		}
 	}
 }
 
+func (wp *wsPeer) handleMessageOfInterest(msg IncomingMessage) bool {
```

Comment:
a bit strange return code: false when processed and error, and true on close. Consider using `error` or naming that bool as *continuation* or something.

Reply:
done.